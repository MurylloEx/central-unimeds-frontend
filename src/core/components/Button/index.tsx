import { Fragment, FunctionComponent } from "react";

export interface ButtonProps { }

export const Button: FunctionComponent<ButtonProps> = () => {
  return (<Fragment />);
}

