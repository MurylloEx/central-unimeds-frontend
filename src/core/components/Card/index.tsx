import { CSSProperties, Fragment, FunctionComponent } from "react";
import { useNavigate } from "react-router-dom";
import { Container } from "./styles";

export interface CardProps {
  image: string;
  routerLink: string;
  params: any;
  style?: CSSProperties | undefined;
}

export const Card: FunctionComponent<CardProps> = ({ image, routerLink, params, style }) => {

  const navigate = useNavigate();

  function onClick(){
    navigate(routerLink, { state: params });
  }

  return (
    <Container 
      style={style}
      onClick={onClick} 
      cardImage={image} 
    />
  );
}

