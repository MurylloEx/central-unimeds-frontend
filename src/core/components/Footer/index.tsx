import { Fragment, FunctionComponent } from "react";

export interface FooterProps { }

export const Footer: FunctionComponent<FooterProps> = () => {
  return (<Fragment />);
}

