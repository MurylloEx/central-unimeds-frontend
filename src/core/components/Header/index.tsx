import { Fragment, FunctionComponent } from "react";

export interface HeaderProps { }

export const Header: FunctionComponent<HeaderProps> = () => {
  return (<Fragment />);
}

