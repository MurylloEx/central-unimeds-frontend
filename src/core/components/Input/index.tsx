import React, { FunctionComponent } from "react";
import { Container, SearchIcon, TextInput } from "./styles";

export interface InputProps {
  onChange?: React.ChangeEventHandler<HTMLInputElement> | undefined;
  value: string;
}

export const Input: FunctionComponent<InputProps> = ({ onChange, value }) => {
  return (
    <Container>
      <SearchIcon />
      <TextInput 
        id="textInput"
        placeholder="Pesquisar" 
        onChange={onChange}
        value={value}
      />
    </Container>
  );
}

