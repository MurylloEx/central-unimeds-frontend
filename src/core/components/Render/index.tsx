import React, { Fragment, FunctionComponent } from "react";

export interface RenderProps {
  if: boolean;
  children: React.ReactNode;
}

export const Render: FunctionComponent<RenderProps> = ({ if: condition, children }: RenderProps) => {
  return condition ? (
    <Fragment>
      {children}
    </Fragment>
  ) : (
    <Fragment />
  );
}