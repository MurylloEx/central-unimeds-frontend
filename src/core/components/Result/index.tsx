import React, { FunctionComponent } from 'react';
import { Container, Row, Title, Text, Globe, Phone, Location, Separator, MetadataBlock } from './styles';
import { capitalize } from '../../services';

export interface ResultProps {
  name: string;
  siteUrl?: string;
  phone: string;
  address: string;
  uf: string;
}

export const Result: FunctionComponent<ResultProps> = (props) => {
  return (
    <Container>
      <Title href={'http://' + props.siteUrl ?? "#notfound"}>{props.name}</Title>
      <Row>
        <MetadataBlock>
          <Globe />
          <Text>{props.siteUrl || "Nenhum site cadastrado"}</Text>
          <Separator /> 
        </MetadataBlock>

        <MetadataBlock>
          <Phone />
          <Text>{props.phone}</Text>
          <Separator />
        </MetadataBlock>
        
        <MetadataBlock>
          <Location />
          <Text>{capitalize(props.address) + ", " + props.uf}</Text>
        </MetadataBlock>
      </Row>
    </Container>
  );
}
