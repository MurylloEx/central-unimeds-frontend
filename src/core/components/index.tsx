export * from './Input';
export * from './Render';
export * from './Result';