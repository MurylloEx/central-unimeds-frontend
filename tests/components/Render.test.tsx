import React from 'react';
import { create } from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Render } from '../../src/core/components';

describe('Render component', () => {

  test('should render without errors the Render component', async () => {
    render(
      <Render if={true}>
        {/* Nothing here */}
      </Render>
    );
  });

  test('should not render the inner components', async () => {
    const { container } = render(
      <Render if={false}>
        <input 
          placeholder="Some placeholder" 
          id="inputId" 
          value="Nothing" 
          onChange={() => {}}
        />
      </Render>
    );

    const input = container.querySelector('#inputId');

    expect(input).toBeNull();
  });

  test('should render the inner components', async () => {
    const { container } = render(
      <Render if={true}>
        <input 
          placeholder="Some placeholder" 
          id="inputId" 
          value="Nothing" 
          onChange={() => {}}
        />
      </Render>
    );

    const input = container.querySelector('#inputId');

    expect(input).not.toBeNull();
  });

  test('should match rendered component with snapshot', () => {
    const tree = create(
      <Render if={true}>
        {/* Nothing here */}
      </Render>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
