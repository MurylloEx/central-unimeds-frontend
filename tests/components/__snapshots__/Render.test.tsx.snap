// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component should match rendered component with snapshot 1`] = `null`;
